.github/workflows/continuous-integration.yml

```diff
@@ -101,10 +103,22 @@ jobs:
       - name: Create test results folder
         run: mkdir -p test-results
 
+      - name: List changed files
+        id: changed-files
+        run: echo ::set-output name=all_changed_files::$(git diff --name-only origin/master...)
+
+      - name: Get app folders for changed files
+        id: get-changed-apps
+        if: ${{ github.ref != 'refs/heads/master' }}
+        run: echo ::set-output name=app_folders::$(node script/github-actions/get-changed-apps.js --output-type folder)
+        env:
+          CHANGED_FILE_PATHS: ${{ steps.changed-files.outputs.all_changed_files }}
+
       - name: Run unit tests
-        run: yarn test:unit --coverage
+        run: yarn test:unit ${APP_FOLDERS:+"{script,$APP_FOLDERS}/**/*.unit.spec.js?(x)"} --coverage
```

Comment:
I'm running the test in the `script` directory (only `github-actions/get-changed-apps.unit.spec.js`) for changed app builds to ensure the script is working as expected. This should help troubleshooting _changed app builds_ if needed by making it easier to rule out the functionality of the script as an issue.

Reply:
This is nice and clean, as planned!

Regarding running the tests in the `script` dir, if any files in that dir are changed, wouldn't that skip changed app builds and run the full test suite anyway?

Reply:
Yeah, files changed in the `script` directory would run a full build. The reason I added this test here (could just be me being overly cautious haha) is because if we ever had an issue with a changed app build, i.e. it didn't select all the right tests for some reason, we could go in GHA and make sure that the unit tests for the `get-changed-apps` script passed. That way we'd have a good sense that the problem may be coming from somewhere else in the process, like maybe some logic in the `select-cypress-tests.js` script was updated, that ended up skipping the mega menu test. There's no reason why the tests for `get-changed-apps` shouldn't pass if the script wasn't updated, but it'd make it easier to know that it's working as intended when troubleshooting.

A situation like that could warrant inspecting the `get-changed-apps` script to make sure it's returning URLs correctly, but if we see that the script unit tests passed in CI, this will make it obvious sooner that the issue isn't coming from the script. But like I said, it could just be me being overly cautious haha.

Reply:
I don't feel too strongly about it, certainly it barely adds any time to run those extra tests in the `script` dir, so there's not much cost to doing it more frequently.

Reply:
Yeah, if for some reason we find that it starts taking extra time from the build, we can always remove it from here. Or maybe changed app builds won't ever have any issues, and we can remove it then too 😅